### DISCUSSION TITLE
Does this plugin work with JSON RPC?

### DISCUSSION THREAD

**Discussion by jacobsiegel1998** [grafana/grafana-json-datasource#136] [Q&A]
Hello, I am trying to get data from an API that is built with JSON-RPC. To get this data I have to POST to the API and the data comes down in the response. Is this possible with this plugin? I have been tinkering with it a bit and I am struggling to find a way to make it work.

**Comment by marcusolsson** [comment]
I'm not very familiar with JSON RPC unfortunately. Could you give me an example of the request you'd like to make, and what response you'd like?

**Reply by jacobsiegel1998** [reply]
All I need to do it make a POST request with a body. Then the API takes that request and body to generate the response. It seems like I'm able to do that in the plugin but I'm getting stuck with it.

**Reply by jacobsiegel1998** [reply]
Here is an example from COSMOS of their API:
```
--> {"jsonrpc": "2.0", "method": "query", "params": [{"start_time_sec": 1514764800, "start_time_usec": 0, "end_time_sec": 415000000, "end_time_usec": 0, "item": ["INST", "HEALTH_STATUS", "TEMP1"], "reduction": "NONE", "value_type": "CONVERTED"}], "id": 1}
<-- {"jsonrpc": "2.0", "result": [[10.3, 1514764800, 0, 1, 1], [15.6, 1514764801, 340, 1, 1]], "id": 1}
```
The first part is the body of the request and then I would want to plot some points from "result"

**Reply by jacobsiegel1998** [reply]
@marcusolsson  anything on this?
